Abort request when returning a completed FlushResult

halter73 opened the pull request:
- This is identical to OnInputOrOutputCompleted firing

Addresses #21834

halter73 committed:
Abort request when returning a completed FlushResult

- This is identical to OnInputOrOutputCompleted firing

halter73 committed:
fixup

halter73 committed:
Add test

jkotalik commented:
@halter73 do you still want to chat about this change?

jkotalik commented on src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs:
```diff
@@ -59,6 +57,7 @@ public Http1Connection(HttpConnectionContext context)
                 _context.ServiceContext.Log,
                 _context.TimeoutControl,
                 this,
+                this,
```
nit: named params here.

halter73 committed:
Address PR feedback

halter73 committed:
Always PoisonBodyStreamsAndPipes on abort

- Fix test that was flaky because of this by waiting for a response
  before closing the connection

halter73 committed:
cleanup
